documents/service.go

```diff
@@ -150,13 +150,13 @@ func (s service) CreateProofsForVersion(ctx context.Context, documentID, version
 	return s.createProofs(model, fields)
 }
 
-func (s service) RequestDocumentSignature(ctx context.Context, model Model) (*coredocumentpb.Signature, error) {
+func (s service) RequestDocumentSignature(ctx context.Context, model Model, senderID identity.DID) (*coredocumentpb.Signature, error) {
 	acc, err := contextutil.Account(ctx)
 	if err != nil {
 		return nil, ErrDocumentConfigAccountID
 	}
 
-	if err := SignatureRequestValidator(s.idService).Validate(nil, model); err != nil {
+	if err := RequestDocumentSignatureValidator(s.idService, senderID).Validate(nil, model); err != nil {
```

Comment:
This will always succeed for transitionValidator since you are not passing the old version of the document.